### PR TITLE
Using ThreadPool to speed up batch pre-processing

### PR DESCRIPTION
**Pull request by lilohuang**
Tasks in pre-processing stage are thread safe. Using ThreadPool to improve the batch pre-processing performance at least 2x faster with large batch size. (e.g. python flow --imgdir sample_img/ --model cfg/yolo.cfg --load bin/yolo.weights --batch 32)

Note: Please evaluate the performance improvement with large number of images instead of 8 images in sample_img/ by default.

### CODE REVIEW

File: darkflow/net/flow.py [hunk 1]
```diff
@@ -115,15 +115,10 @@ def predict(self):
...
-        inp_feed = list(); new_all = list()
         this_batch = all_inps[from_idx:to_idx]
-        for inp in this_batch:
-            new_all += [inp]
-            this_inp = os.path.join(inp_path, inp)
-            this_inp = self.framework.preprocess(this_inp)
-            expanded = np.expand_dims(this_inp, 0)
-            inp_feed.append(expanded)
-        this_batch = new_all
+        inp_feed = pool.map(lambda inp: (
```
Comment (thtrieu) [inline darkflow/net/flow.py]: Where is this package pool comes from? do we have to import it? install?

Comment (lilohuang) [reply darkflow/net/flow.py]: pool variable was initialized at the beginning of flow.py which was already used in ThreadPool batch output few months ago.

Comment (lilohuang) [reply darkflow/net/flow.py]: pool variable is initialized to Python built-in ThreadPool at the beginning of flow.py which has been used in ThreadPool batch output for prediction few months ago. 😎

Comment (lilohuang) [reply darkflow/net/flow.py]: "pool" variable is initialized as a Python built-in ThreadPool instance at flow.py#L15 which has been used in ThreadPool batch output for prediction few months. 😎

Comment (lilohuang) [reply darkflow/net/flow.py]: "pool" variable is initialized as a Python built-in ThreadPool instance at flow.py#L15 which has been used few months in ThreadPool batch post-processing for prediction. 😎

**Pull request merged**